backend/mem/fileSystem.go

```diff
@@ -0,0 +1,78 @@
+package mem
+
+import (
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/backend"
+	"github.com/c2fo/vfs/v4/utils"
+	"path"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+var systemMap map[string]*File
+var fileList []*File
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
+type FileSystem struct{}
+
+// FileSystem will return a retrier provided via options, or a no-op if none is provided.
+func (fs *FileSystem) Retry() vfs.Retry {
+	return vfs.DefaultRetryer()
+}
+
+// NewFile function returns the mem implementation of vfs.File.  NOT DONE
+func (fs *FileSystem) NewFile(volume string, name string) (vfs.File, error) {
+
+	if !path.IsAbs(name) {
+		name = path.Join("/", name)
+	}
+
+	file, _ := newFile(name)
+	tmp, err := fs.NewLocation(volume, name)
```

Comment:
Here you're using name... if name was /some/path/to/file.txt you're passing that to the NewLocation() when all you should be passing is path.Dir(name)